src/py/flwr/driver/serde.py

```diff
@@ -15,35 +15,161 @@
 """Flower driver serialization."""
 
 
-from flwr.driver.messages import GetClientsRequest, GetClientsResponse
-from flwr.proto import driver_pb2
+from typing import List
+
+from flwr.common.serde import (
+    client_message_from_proto,
+    client_message_to_proto,
+    server_message_from_proto,
+    server_message_to_proto,
+)
+from flwr.driver.messages import (
+    CreateTasksRequest,
+    CreateTasksResponse,
+    GetClientsRequest,
+    GetClientsResponse,
+    GetResultsRequest,
+    GetResultsResponse,
+    Result,
+    Task,
+    TaskAssignment,
+)
+from flwr.proto import driver_pb2, task_pb2
+
+# === GetClients messages ===
 
 
 def get_clients_request_to_proto(
     req: GetClientsRequest,
 ) -> driver_pb2.GetClientsRequest:
-    """."""
+    """Serialize `GetClientsRequest` to ProtoBuf."""
     # pylint: disable=W0613
     return driver_pb2.GetClientsRequest()
 
 
 def get_clients_request_from_proto(
     msg: driver_pb2.GetClientsRequest,
 ) -> GetClientsRequest:
-    """."""
+    """Deserialize `GetClientsRequest` from ProtoBuf."""
     # pylint: disable=W0613
     return GetClientsRequest()
 
 
 def get_clients_response_to_proto(
     res: GetClientsResponse,
 ) -> driver_pb2.GetClientsResponse:
-    """."""
+    """Serialize `GetClientsResponse` to ProtoBuf."""
     return driver_pb2.GetClientsResponse(client_ids=res.client_ids)
 
 
 def get_clients_response_from_proto(
     msg: driver_pb2.GetClientsResponse,
 ) -> GetClientsResponse:
-    """."""
+    """Deserialize `GetClientsResponse` from ProtoBuf."""
     return GetClientsResponse(client_ids=list(msg.client_ids))
+
+
+# === CreateTasks messages ===
+
+
+def create_tasks_request_to_proto(
+    req: CreateTasksRequest,
+) -> driver_pb2.CreateTasksRequest:
+    """Serialize `CreateTasksRequest` to ProtoBuf."""
+    task_assignments_proto: List[task_pb2.TaskAssignment] = []
+    for task_assignment in req.task_assignments:
+        task_assignment_proto = task_pb2.TaskAssignment(
+            task=task_pb2.Task(
+                task_id=task_assignment.task.task_id,
+                legacy_server_message=server_message_to_proto(
+                    task_assignment.task.legacy_server_message
+                ),
+            ),
+            client_ids=task_assignment.client_ids,
+        )
```

Comment:
Nitpick:
I just thought it might be more readable if we reduced the nesting a bit. Wdyt?


```suggestion
        legacy_server_message_proto=server_message_to_proto(
            task_assignment.task.legacy_server_message
        ),
        task_proto=task_pb2.Task(
            task_id=task_assignment.task.task_id,
            legacy_server_message=legacy_server_message_proto
        ),
        task_assignment_proto = task_pb2.TaskAssignment(
            task=task_proto
            client_ids=task_assignment.client_ids,
        )
```

Reply:
@tanertopal I agree, it's done for this and other functions in the latest commit